src/courses/views.py

```diff
@@ -44,12 +44,20 @@ def mark_calculations(request, user_id=None):
     else:
         xp_per_course = None
 
+    # only show mark ranges where student is enrolled in and is also active
+    user_courses = [sc.course for sc in user.coursestudent_set.all()]
```

Comment:
I believe there is already a method in the Profile model to get a student's current courses.  Since that will already be fully tested, better to use that, also easier to read the code:  `user.profile.current_courses()`.

Reply:
seems like current_courses() only returns StudentCourses

so its still roughly the same

user_courses = [sc.course for sc in user.profile.current_courses()]

Reply:
changed

Reply:
Ah yes, I see what you mean.  You should use [values_list](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#values-list) on the queryset (much better habit to stick with queryset operations when you can, which are way more efficient):
`user.profile.current_courses().values_list('course', flat=True)` <-- gives you a list of the provided field.  End result should be identical to your list comprehension.

Reply:
tweaked